### DISCUSSION TITLE
Sparql update produces ParseError which I cannot comprehend.

### DISCUSSION THREAD

**Discussion by XBagon** [oxigraph/oxigraph#452] [Q&A] [answered]
My update looks like this:
```sparql
PREFIX : <http://myprefix.com/>
DELETE { ?neighbor :last_seen ?last_seen. }
INSERT { ?neighbor :last_seen ?new_last_seen. }
WHERE { ?neighbor :last_seen ?last_seen. ?neighbor :entityId 0. :cg :age ?age. BIND(?age AS ?new_last_seen). };
DELETE { :cg :age ?age. }
INSERT { :cg :age ?new_age. }
WHERE { :cg :age ?age. BIND((?age + 1) AS ?new_age). }
```

and for some reason I get this error: 
```rs
Parsing(ParseError { inner: Parser(ParseError { location: LineCol { line: 4, column: 74, offset: 206 }, expected: ExpectedSet { expected: {"OPTIONAL"} } }) })
```
which tells me to put an OPTIONAL where the triple starting with `:cg` is and I don't know why? But also when I change the order and put the `?neighbor :entityId 0` triple to the end of the WHERE clause, it doesn't throw an error, but sadly also doesn't update any values.

I'm not sure if this is me not understanding something and if it has anything to do with this specific implementation, but I would be thankful for any ideas regardless :p.

**Comment by Tpt** [comment, marked as answer]
Hi! Thank you for using Oxigraph and sorry for the very very bad error message. This is something I should definitely fix. The parser is failing because `0.` is parsed as the decimal `0.` and not the integer `0` followed by a `.`. I plan to fix that in the next major Oxigraph release (#375). Sorry again for that.

**Reply by XBagon** [reply]
Hi! Thanks for the quick reply & solution :)! 
Don't worry, looking at it now, it seems kinda obvious. Also I'm very happy Oxigraph exists, was rather surprised to find a SPARQL db implementation of this quality :)
